transcom/mymove
# B 21533 int HQ role with Safety Move privs

## [B-21533](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1021039)

## Summary

Previously code was added to prevent HQ role from being able to have Safety Move privileges, but it was decided that HQ should now have access to see Safety Moves. This PR removes the checks preventing the HQ role from selecting the Safety Move priv in the admin console.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

To be able to select Safety Moves you need to first add the Super permissions for the admin account. This can be done by changing the flag in the DB for `super` to true in the `admin_users` table.

![image](https://github.com/user-attachments/assets/d4067584-6ebc-4e4b-b69a-72bfd717656c)

Now you can either edit and existing Office User and give them Headquarters Role and Safety Move privileges or create a new Office User with the HQ role and Safety Move priv. 

NOTE: If creating a new Office User you first must select the HQ role and save and then you can add the Safety Move privilege. This is because the checkbox for Safety Move is only currently available on the edit Office User page and not create page (known issue). 

1. With an admin who has super powers (DB flag see above) access the admin app
2. Create or Edit an existing Office User and give them at least the Service Counselor and Headquarters Role and Safety Move privileges. Make sure to remember which GBLOC you gave the Office User.
3. While viewing the create or edit Office User page you should see the following label:
### The Safety Moves privilege can only be selected for the following roles: Task Ordering Officer, Task Invoicing Officer, Services Counselor, Quality Assurance Evaluator, Customer Service Representative, and Headquarters
4. Verify you can select Safety Moves privileges with the following roles and any combination of: 

Task Ordering Officer
Task Invoicing Officer
Services Counselor
Quality Assurance Evaluator
Customer Service Representative
Headquarters

5. Verify you can NOT select Safety Move privileges with the following roles and any combination of:

Customer
Contracting Officer
Prime Simulator
Government Surveillance Representative

6. Login to the Office App as the Service Counselor, go to the Customer Search and type in any and perform the search
7. Click the Add Customer button
8. When creating the customer you will be asked if this is a Safety Move, select Yes
9. Finish filling out the fields and create a new move
10. Go through and create an HHG shipment, make sure you set it up so that your shipment falls under the same GBLOC as your Office User so you will see the move in your Queue
11. Make sure you enter all the Order information but don't submit the shipment yet, copy your move code
12. Change your user role to Headquarters and go to the Counseling Queue
13. Verify you can see your Safety Move (should have a ${\color{red}Safety}$ label and move code starts with `SM`)
14. Select the Safety Move and verify you can view the shipment
15. Switch the role back to Service Counselor and Submit the shipment
16. Change back to the Headquarters role and verify you can see the shipment in the  TOO queue
17. Select the shipment and verify you can view it

### Frontend
